src/main/scala/is/hail/variant/VariantDataset.scala

```diff
@@ -176,9 +190,19 @@ object VariantDataset {
       case _ => false
     }
 
+    val isGenericGenotype = fields.get("isGenericGenotype") match {
+      case Some(t: JBool) => t.value
+      case Some(other) => fatal(
+        s"""corrupt VDS: invalid metadata
+           |  Expected `JBool' in field `isGenericGenotype', but got `${ other.getClass.getName }'
+           |  Recreate VDS with current version of Hail.""".stripMargin)
+      case _ => false
+    }
+
     val saSignature = Parser.parseType(getAndCastJSON[JString]("sample_annotation_schema").s)
     val vaSignature = Parser.parseType(getAndCastJSON[JString]("variant_annotation_schema").s)
     val globalSignature = Parser.parseType(getAndCastJSON[JString]("global_annotation_schema").s)
+    val genotypeSignature = Parser.parseType(getAndCastJSON[JString]("genotype_schema").s)
```

Comment:
This will fail loading an old VDS.  You have to make it optional default to TGenotype.